### PR TITLE
Fix benchmarks failing on scheduler overflow

### PR DESCRIPTION
**Pull request by muharem**
Scheduler has the upper limit for the number of the calls scheduled per block.
The benchmarks of the `pallet_referenda` and `pallet_conviction_voting` do not respect the limit and overflows it if the scheduler `MaxScheduledPerBlock` not high enough.

For `pallet_referenda` benchmarks the limit reached **if**
track.max_deciding > scheduler.max_scheduled_per_block **OR** 
referenda.max_queued > scheduler.max_scheduled_per_block **OR** 
track.max_deciding + referenda.max_queued > scheduler.max_scheduled_per_block 

For `pallet_conviction_voting` benchmarks the limit reached **if**
number_of_registered_votes > scheduler.max_scheduled_per_block 

This PR fixes it with aim to make it seamless for users.

### CODE REVIEW

File: frame/support/src/traits/schedule.rs [hunk 1]
```diff
@@ -426,6 +426,10 @@ pub mod v3 {
 		///
 		/// Will return an `Unavailable` error if the `address` is invalid.
 		fn next_dispatch_time(address: Self::Address) -> Result<BlockNumber, DispatchError>;
+
+		/// Return the maximum number of tasks possible to schedule per block.
+		#[cfg(feature = "runtime-benchmarks")]
+		fn max_scheduled_per_block() -> u32;
```
Comment (muharem) [inline frame/support/src/traits/schedule.rs:432]: I don't see a better solution, how to access to this configuration from the referenda benchmarks (and keep it seamless for the client). Any ideas?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2267092

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2267132

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2267133

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2267134

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2267228

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2267229

**Comment by muharem** [comment]
I did not expect the solution to be so complex, here the alternative way that only helps a user to find a workaround https://github.com/paritytech/substrate/pull/13231
cc: @bkchr @ggwpez

**Pull request closed without merging**